rest-api/src/main/java/com/cloudbees/workflow/rest/external/RunExt.java

```diff
@@ -209,8 +248,92 @@ public static RunExt create(WorkflowRun run) {
             if (artifacts != null && !artifacts.isEmpty()) {
                 runExt.get_links().setArtifacts(Link.newLink(RunAPI.getArtifactsUrl(run)));
             }
+        }
+        return runExt;
+    }
+
+    /** Creates a wrapper of this that hides the full stage nodes
+     *  Use case: returning a minimal view of the run, while using a cached, fully-realized version
+     */
+    public RunExt createWrapper() {
+        return new ChildHidingWrapper(this);
+    }
 
-            FlowGraphWalker walker = new FlowGraphWalker(run.getExecution());
+    protected static class ChildHidingWrapper extends RunExt {
+        protected RunExt myRun;
+        protected List<StageNodeExt> wrappedStages;
+
+        public RunLinks get_links() {return myRun.get_links();}
+        public String getId() {return myRun.getId();}
+        public String getName() {return myRun.getName();}
+        public StatusExt getStatus() {return myRun.getStatus();}
+        public long getStartTimeMillis() {return myRun.getStartTimeMillis();}
+        public long getEndTimeMillis() {return myRun.getEndTimeMillis();}
+        public long getDurationMillis() {return myRun.getDurationMillis();}
+        public long getQueueDurationMillis() {return myRun.getQueueDurationMillis();}
+        public long getPauseDurationMillis() {return myRun.getPauseDurationMillis();}
+        public List<StageNodeExt> getStages() {return Collections.unmodifiableList(wrappedStages);}
+
+        protected ChildHidingWrapper(RunExt run) {
+            this.myRun = run;
+            List<StageNodeExt> myWrappedStages = new ArrayList<StageNodeExt>();
+            if (wrappedStages == null) {
+                for(StageNodeExt stage : run.getStages()) {
+                    myWrappedStages.add(stage.myWrapper());
+                }
+                this.wrappedStages = myWrappedStages;
+            } else {
+                wrappedStages = null;
+            }
+        }
```

Comment:
AFAICT this does not work: `wrappedStages` is always initialized as a copy of `run.stages`, meaning nothing is in fact hidden.